src/cryptographic_primitives/commitments/hash_commitment.rs

```diff
@@ -33,20 +34,55 @@ impl Commitment for HashCommitment {
 
         BigInt::from(digest.finish().as_ref())
     }
+
+    fn create_commitment(
+        message: &BigInt, security_bits: &usize) -> (BigInt, BigInt)
+    {
+        let mut digest = Context::new(&SHA256);
+        let bytes_message: Vec<u8> = message.into();
+        digest.update(&bytes_message);
+        let blinding_factor = &(BigInt::sample(security_bits.clone()));
+        let bytes_blinding_factor: Vec<u8> = blinding_factor.into();
+        digest.update(&bytes_blinding_factor);
+
+        (BigInt::from(digest.finish().as_ref()), blinding_factor.clone())
+    }
 }
 
 #[cfg(test)]
 mod tests {
     use ::BigInteger as BigInt;
     use super::Commitment;
     use super::HashCommitment;
+    use arithmetic::traits::Samplable;
+
 
     #[test]
-    // Very basic test here, TODO: suggest better testing
-    fn create_hash_commitment_test() {
-        let commitment = HashCommitment::create_commitment_with_user_defined_randomness(
-            &BigInt::one(), &BigInt::zero());
+    fn hash_commitment_test() {
+        let sec_bits = 256;
+        let message = BigInt::sample(sec_bits.clone());
+        let (commitment, blind_factor) = HashCommitment::create_commitment(&message, &sec_bits);
+        let commitment2 = HashCommitment::create_commitment_with_user_defined_randomness(
+            &message, &blind_factor);
+        //test commitment length  - works because SHA256 output length the same as sec_bits
+        assert_eq!(commitment.bit_length(),sec_bits);
```

Comment:
Can we create 2 tests, one for create_commitment and one for create_commitment_with_user_defined_randomness that will have a name which self explained the purpose: "create_commitment_bit_length_test" and "create_commitment_with_user_defined_randomness_bit_length_test" and move the code accordingly. It is always better to localized your unit tests as much as you can.

Also I suggest to remove print statements (even if commented).